File: app/Models/Plan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

/**
 * Plan Model — AIVidCatalog18
 *
 * Represents a subscription plan (e.g., Basic $9.99/30 days, Premium $19.99/30 days).
 * Plans define pricing, duration, and access level for the platform.
 *
 * Strictly fictional AI-generated content — no illegal/prohibited material.
 *
 * @property int $id
 * @property string $name
 * @property string $slug
 * @property float $price_usd
 * @property int $duration_days
 * @property string $description
 * @property bool $is_active
 * @property \Carbon\Carbon $created_at
 * @property \Carbon\Carbon $updated_at
 */
class Plan extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'slug',
        'price_usd',
        'duration_days',
        'description',
        'is_active',
    ];

    protected function casts(): array
    {
        return [
            'price_usd'     => 'decimal:2',
            'duration_days' => 'integer',
            'is_active'     => 'boolean',
        ];
    }

    // =========================================================================
    // Relationships
    // =========================================================================

    /**
     * Get all subscriptions for this plan.
     */
    public function subscriptions(): HasMany
    {
        return $this->hasMany(Subscription::class);
    }

    /**
     * Get all payments associated with this plan.
     */
    public function payments(): HasMany
    {
        return $this->hasMany(Payment::class);
    }

    // =========================================================================
    // Scopes
    // =========================================================================

    /**
     * Scope to only active plans available for purchase.
     */
    public function scopeActive($query)
    {
        return $query->where('is_active', true);
    }

    // =========================================================================
    // Helpers
    // =========================================================================

    /**
     * Format the price for display (e.g., "$9.99").
     */
    public function formattedPrice(): string
    {
        return '$' . number_format($this->price_usd, 2);
    }
}

File: mobile/lib/models/plan.dart
/// Plan Model — AIVidCatalog18 Flutter App
///
/// Subscription plan data from the API.
/// Strictly fictional AI-generated content — no illegal/prohibited material.

class Plan {
  final int id;
  final String name;
  final String slug;
  final double priceUsd;
  final int durationDays;
  final String description;
  final bool isActive;

  Plan({
    required this.id,
    required this.name,
    required this.slug,
    required this.priceUsd,
    required this.durationDays,
    required this.description,
    required this.isActive,
  });

  factory Plan.fromJson(Map<String, dynamic> json) {
    return Plan(
      id: json['id'] as int,
      name: json['name'] as String,
      slug: json['slug'] as String,
      priceUsd: double.tryParse(json['price_usd'].toString()) ?? 0.0,
      durationDays: json['duration_days'] as int? ?? 30,
      description: json['description'] as String? ?? '',
      isActive: json['is_active'] as bool? ?? true,
    );
  }

  /// Formatted price string
  String get formattedPrice => '\$${priceUsd.toStringAsFixed(2)}';

  /// Is this the premium plan?
  bool get isPremium => slug == 'premium';
}
